src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxFactoryTests.cs

```diff
@@ -83,6 +83,25 @@ public void TestConstructClassWithKindAndNameOnly()
             Assert.Equal(SyntaxKind.None, c.SemicolonToken.Kind());
         }
 
+        [Fact]
+        public void IndexerDeclarationUpdateRetainsSemicolonToken()
+        {
+            var comment = SyntaxFactory.Comment("//original");
+            SyntaxToken originalSemicolonToken = SyntaxFactory.Token(SyntaxKind.SemicolonToken).WithLeadingTrivia(comment);
+            var initialDeclaration = SyntaxFactory.IndexerDeclaration(SyntaxFactory.ParseName("System.String"))
+                .WithSemicolonToken(originalSemicolonToken);
+
+            var mutatedDeclaration = initialDeclaration.Update(initialDeclaration.AttributeLists,
+                initialDeclaration.Modifiers,
+                initialDeclaration.Type,
+                initialDeclaration.ExplicitInterfaceSpecifier,
+                initialDeclaration.ThisKeyword,
+                initialDeclaration.ParameterList,
+                initialDeclaration.AccessorList);
+
+            Assert.Contains(comment.ToString(), mutatedDeclaration.SemicolonToken.GetLeadingTrivia().ToFullString());
```

Comment:
In general, we prefer doing something like `AssertEx.Equals("text", mutatedDeclaration.SemicolonToken.GetLeadingTrivia().ToFullString())`. It makes it easier to read the test at the cost of a very minor, local duplication, which is worth it imo.